File: backend-ruby/app/models/diem_danh.rb
class DiemDanh < ApplicationRecord
  self.table_name = 'DiemDanh'
  self.primary_key = 'DiemDanhID'
  belongs_to :ky_thi, class_name: "KyThi", foreign_key: "KyThiID"
  belongs_to :phong_thi, class_name: "PhongThi", foreign_key: "PhongThiID"
  belongs_to :nguoi_ghi_nhan, class_name: "NguoiDung", foreign_key: "NguoiGhiNhanID", optional: true
  belongs_to :dang_ky_thi, class_name: "DangKyThi", foreign_key: "DangKyThiID"
end

File: backend-csharp/Models/DiemDanh.cs
namespace backend_csharp.Models;

public class DiemDanh
{
    public int DiemDanhID { get; set; }
    public int DangKyThiID { get; set; }
    public int KyThiID { get; set; }
    public int PhongThiID { get; set; }
    public string TrangThai { get; set; } = "absent";
    public DateTime? ThoiGianCheckIn { get; set; }
    public int? NguoiGhiNhanID { get; set; }
    public string? GhiChu { get; set; }
    public DateTime TaoLuc { get; set; }
    public DateTime? CapNhatLuc { get; set; }
    public KyThi? KyThi { get; set; }
    public PhongThi? PhongThi { get; set; }
    public NguoiDung? NguoiGhiNhan { get; set; }
    public DangKyThi? DangKyThi { get; set; }
}
